src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -341,67 +345,6 @@ JNI_METHOD(void, deleteDeviceController)(JNIEnv * env, jobject self, jlong devic
     }
 }
 
-void HandleSimpleOperationComplete(ChipDeviceController * deviceController, void * appReqState)
```

Comment:
How's removing this related to the Java compiling?

Reply:
GN compiler is more strict than autotools and yells at me for 'unused static function'. Figured I would drop it instead of supressing the warning.

Same with the pthread bit - unsure why autotools works but pegged at different system (clang vs gcc) and added the define like we have in SystemLayer.cpp.